protocols/migration/migration_reports_5_0_0_to_reports_4_0_0.py

```diff
@@ -4,14 +4,25 @@
 from protocols import reports_5_0_0 as reports_5_0_0
 from protocols.migration.base_migration import BaseMigration
 from protocols.migration.base_migration import MigrationError
+from protocols.migration.participants import MigrationParticipants103To100
 from protocols.migration.migration_participant_1_1_0_to_participant_1_0_0 import MigrateParticipant110To100
+from protocols.migration.migration_participant_1_1_0_to_participant_1_0_3 import MigrateParticipant110To103
 
 
 class MigrateReports500To400(BaseMigration):
 
     old_model = reports_5_0_0
     new_model = reports_4_0_0
 
+    tier_map = {
```

Comment:
is this actually needed at all given they are the same strings?

Reply:
I'll answer myself: it is not. See:
```
    def convert_class(target_klass, instance):
        new_instance = target_klass.fromJsonDict(
            jsonDict=instance.toJsonDict()
        )
        return new_instance
```

Reply:
@priesgo it is needed because TIER4 and TIER5 doesn't exit in the older model.
@kevinpetersavage I'll use the mapping in the base class when it's merged.

Reply:
oh I see, not really, what you need is to use a factory to generate appropriate tiers. I thought I commented this yesterday.
```
valid_tiers = [self.old_model.Tier.TIER1, self.old_model.Tier.TIER2,
                       self.old_model.Tier.TIER3, self.old_model.Tier.NONE]

        # registers factories to ensure that tiers in report events are only valid ones
        class ReportEventFactoryNulls(FactoryAvro):
            def __init__(self, *args, **kwargs):
                super(ReportEventFactoryNulls, self).__init__(*args, **kwargs)

            class Meta:
                model = self.old_model.ReportEvent
            _version = VERSION_61
            _fill_nullables = True
            tier = factory.fuzzy.FuzzyChoice(valid_tiers)
        GenericFactoryAvro.register_factory(self.old_model.ReportEvent, ReportEventFactoryNulls, VERSION_61, True)

        class ReportEventFactory(FactoryAvro):
            def __init__(self, *args, **kwargs):
                super(ReportEventFactory, self).__init__(*args, **kwargs)

            class Meta:
                model = self.old_model.ReportEvent
            _version = VERSION_61
            _fill_nullables = False
            tier = factory.fuzzy.FuzzyChoice(valid_tiers)
        GenericFactoryAvro.register_factory(self.old_model.ReportEvent, ReportEventFactory, VERSION_61, False)
```

Can you add to the base class?